core/unit_test/TestViewTypedefs.cpp

```diff
@@ -92,7 +95,7 @@ constexpr bool test_view_typedefs_impl() {
   // =========================================
   static_assert(std::is_same_v<typename ViewType::traits, ViewTraitsType>);
   static_assert(std::is_same_v<typename ViewType::array_type,
-                               Kokkos::View<typename ViewType::scalar_array_type, typename ViewType::array_layout,
+                               Kokkos::View<typename ViewType::data_type, typename ViewType::array_layout,
```

Comment:
This needs to be changed back with deprecation guard, when the "array_type" typedef is changed back and fixed.